ggml-cuda.cu

```diff
@@ -6601,8 +6601,8 @@ static __global__ void flash_attn_ext_f16(
                     smax = warp_reduce_max(__hmax(smax, s));
                     M[j] = warp_reduce_max(__hmax(M[j], s));
 
-                    const half ms = __hisinf(m) ? __float2half(0.0f) : hexp(m - M[j]);
-                    const half vs = __hisinf(s) ? __float2half(0.0f) : hexp(s - M[j]);
+                    const half ms = __hisinf(m) == -1 ? __float2half(0.0f) : hexp(m - M[j]);
+                    const half vs = __hisinf(s) == -1 ? __float2half(0.0f) : hexp(s - M[j]);
```

Comment:
Hm, is this correct? `__hisinf` returns bool:

https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH____HALF__COMPARISON.html#group__CUDA__MATH____HALF__COMPARISON_1gc6698aa5f990114a7bcd129ed9375f17

Reply:
![Screenshot_20240202-080354_Chrome.jpg](https://github.com/Pints-App/llama.cpp/assets/57494570/3144ece6-e9d1-424c-988f-2447f21570be)

Returns an int (0 not infinite, 1 positive infinite, -1 negative infinite)

Reply:
Oh wow 🤦 - you are correct